### PR TITLE
Add one storage case HotAdd_SCSI_Dynamic ESX-STOR-001

### PR DESCRIPTION
**Pull request by lixuemin2016**
<test>
            <testName>HotAdd_SCSI_Dynamic</testName>
            <setupScript>SetupScripts\add_hard_disk.ps1</setupScript>
            <testScript>stor_lis_disk.sh</testScript>
            <files>remote-scripts/stor_lis_disk.sh,remote-scripts/utils.sh,
            remote-scripts/stor_utils.sh </files>
            <cleanupScript>SetupScripts\remove_hard_disk.ps1</cleanupScript>
            <RevertDefaultSnapshot>True</RevertDefaultSnapshot>
            <timeout>1800</timeout>
            <testparams>
                <param>DiskType=SCSI</param>
                <param>StorageFormat=Thin</param>
                <param>CapacityGB=3</param>
                <param>fileSystems=(ext4 ext3 xfs btrfs)</param>
                <param>TC_COVERED=ESX-STOR-001</param>
            </testparams>
            <onError>Continue</onError>
        </test>

### CODE REVIEW

**Comment by lixuemin2016** [comment]
<test>
            <testName>HotAdd_SCSI_Dynamic_2T</testName>
            <setupScript>SetupScripts\add_hard_disk.ps1</setupScript>
            <testScript>stor_lis_disk.sh</testScript>
            <files>remote-scripts/stor_lis_disk.sh,remote-scripts/utils.sh,
            remote-scripts/stor_utils.sh </files>
            <cleanupScript>SetupScripts\remove_hard_disk.ps1</cleanupScript>
            <RevertDefaultSnapshot>True</RevertDefaultSnapshot>
            <timeout>1800</timeout>
            <testparams>
                <param>DiskType=SCSI</param>
                <param>StorageFormat=Thin</param>
                <param>CapacityGB=2050</param>
                
                <param>fileSystems=(ext4 xfs)</param>
                <param>diskFormatType=parted</param>
                <param>TC_COVERED=ESX-STOR-003</param>
            </testparams>
            <onError>Continue</onError>
        </test>

**Comment by lixuemin2016** [comment]
<test>
            <testName>VHDx_Mount_NFS_server</testName>
            <setupScript>SetupScripts\setup_nfs_server.ps1</setupScript>
            <testScript>testscripts\stor_nfs_client.ps1</testScript>
            <files>remote-scripts/utils.sh,
            remote-scripts/stor_utils.sh </files>
            <RevertDefaultSnapshot>True</RevertDefaultSnapshot>
            <timeout>1800</timeout>
            <testparams>
                <param>TC_COVERED=ESX-STOR-002</param>
            </testparams>
            <onError>Continue</onError>
        </test>

File: remote-scripts/stor_lis_disk.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    UpdateSummary "Error in check disk size"
+    SetTestStateFailed
+    exit 1
+else
+    LogMsg "disk size check successfully "
+fi
+
+# If does not define file system type, use ext3 as default.
+if  [ ! ${fileSystems} ];then
+    fileSystems=(ext3)
```
Comment (VirtQE-S1) [inline remote-scripts/stor_lis_disk.sh]: ext4 as default disk format

File: remote-scripts/stor_utils.sh [hunk 2]
```diff
@@ -9,18 +9,19 @@
 ##
 ## Revision:
 ## v1.0 - xuli - 09/01/2017 - Draft script for case stor_utils.sh
```
Comment (VirtQE-S1) [inline remote-scripts/stor_utils.sh]: mm/DD/YYYY

File: remote-scripts/stor_lis_disk.sh [hunk 3]
```diff
@@ -66,7 +66,14 @@ fi
 if  [ ! ${fileSystems} ];then
     fileSystems=(ext3)
```
Comment (VirtQE-S1) [inline remote-scripts/stor_lis_disk.sh]: ext4 as default disk format

File: remote-scripts/stor_utils.sh [hunk 4]
```diff
@@ -10,6 +10,9 @@
...
```
Comment (VirtQE-S1) [inline remote-scripts/stor_utils.sh]: mm/DD/YYYY

File: remote-scripts/stor_utils.sh [hunk 5]
```diff
@@ -88,6 +91,27 @@ DoFdisk()
...
+{
+    ############################################################################
+    # Description:
+    #   Use parted /dev/sdb to create partition /dev/sdb1
+    # Parameters:
+    # $1 : storage device name /dev/sdb
+    # Return: 0 : if parted successfully, otherwise, return 1
+    ############################################################################
+    local driveName=$1
+    parted -s -- $driveName mklabel gpt
```
Comment (VirtQE-S1) [inline remote-scripts/stor_utils.sh]: check $?

File: remote-scripts/stor_lis_nfs.sh [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+#Check for NFS_Path
+if [ ! ${NFS_Path} ]; then
+    LogMsg "Error: The NFS_Path variable is not defined."
+    UpdateSummary "Error: The NFS_Path variable is not defined."
+    SetTestStateAborted
+    exit 1
+fi
+
+#Restart nfs service
+service nfs restart
```
Comment (VirtQE-S1) [inline remote-scripts/stor_lis_nfs.sh]: remove nfs restart

File: remote-scripts/stor_utils.sh [hunk 7]
```diff
@@ -0,0 +1,411 @@
...
+    if [ "$?" = "0" ]; then
+        LogMsg "Drive mounted successfully..."
+        return 0
+    else
+        LogMsg "Error in mount ${driveName}1 with $mountPoint .."
+        return 1
+    fi
+}
+
+DoDDFile()
```
Comment (VirtQE-S1) [inline remote-scripts/stor_utils.sh]: check DDed file existed.

File: remote-scripts/stor_lis_nfs.sh [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    LogMsg "Error in restart nfs..."
+    UpdateSummary "Error in restart nfs..."
+    SetTestStateFailed
+    exit 1
+fi
+
+#Mount nfs_path to /mnt
+DoMountFs $NFS_Path $mountPoint $mountType
+if [ "$?" = "0" ]; then
+    LogMsg "mount nfs path successfully "
```
Comment (VirtQE-S1) [inline remote-scripts/stor_lis_nfs.sh]: add updatesummary in both if and else.

File: setupscripts/setup_nfs_server.ps1 [hunk 9]
```diff
@@ -111,25 +111,38 @@ if (-not $vmObj)
     return $false
 }
 
+$snapsOut = Get-Snapshot -VM $vmObj
+
+Set-VM -VM $vmObj -Snapshot $snapsOut.Name -Confirm:$False
```
Comment (VirtQE-S1) [inline setupscripts/setup_nfs_server.ps1]: remove .Name

File: setupscripts/add_hard_disk.ps1 [hunk 10]
```diff
@@ -34,6 +34,7 @@
 
     <test>
         <testName>HotAdd_SCSI_Dynamic</testName>
+        <testID>ESX-STOR-001</testID>
```
Comment (VirtQE-S1) [inline setupscripts/add_hard_disk.ps1]: testID

**Pull request merged**